### ISSUE TITLE
Center network in rmarkdown slides

### ISSUE THREAD

**Issue by neobernad** [christophergandrud/networkD3#278]
Hi,

I'm using `rmarkdown` with `revealjs` to create HTML slides. When I run the following code:

```r
testNet <- forceNetwork(Links = MisLinks, Nodes = MisNodes, Source = "source",
             Target = "target", Value = "value", NodeID = "name",
             Group = "group", opacity = 0.4, arrows = TRUE)
onRender(testNet , customJS)
```
The method `customJS` comes from [this answer](https://stackoverflow.com/questions/56202858/networkd3-package-show-node-names-of-all-connected-nodes-when-hovering-over), which basically shows the label of the connected nodes when hovering the mouse on them (this does not affect my issue, just wanted to clarify what the code does).

The issue is that the graph shows on the left corner of the slide:
![before](https://user-images.githubusercontent.com/47330471/102337459-c8b4cf80-3f92-11eb-8b11-9789c301ea7e.png)

But when I refresh with F5 the slide of the graph, then it is rendered properly:
![after](https://user-images.githubusercontent.com/47330471/102337407-b8045980-3f92-11eb-8072-9236e0d90afc.png)

Is there anyway to force-refresh the network, force the starting (X,Y) node position or add some margins so that the graph shows up in the center?

Best,
José Antonio

**Comment by cjyetman** [comment]
I definitely need a minimal reproducible example if you want me to look into this.

**Comment by cjyetman** [comment]
also, please do not include custom stuff that does not affect your issue in the **minimal** reproducible example

**Comment by neobernad** [comment]
Hello,

Thanks for your fast response. I kept doing some search about how to fix this issue, it seems to be an issue with `revealjs` and `htmlwidget` instead of with `networkD3`, sorry for opening up the issue on your side.

Solution: I fixed it by saving the widget with `saveWidget` method and then using an HTML `<iframe>` to embed the graph.
Source: https://github.com/rstudio/revealjs/issues/11

Thanks anyway for your help!
